crowbar_framework/app/models/deployer_service.rb

```diff
@@ -138,4 +138,21 @@ def transition(inst, name, state)
     Rails.logger.debug("Deployer transition: exiting: #{name} for #{state}")
     return [200, { name: name }]
   end
+
+  # try to know if we can skip a node from running chef-client
+  def skip_unchanged_node?(node_name, old_role, new_role)
+    # if old_role is nil, then we are applying the barclamp for the first time
+    return false if old_role.nil?
+
+    # if the node changed roles, then we need to apply
+    return false if node_changed_roles?(node_name, old_role, new_role)
+
+    # if attributes have changed, we need to run
+    return false if node_changed_attributes?(node_name, old_role, new_role)
```

Comment:
That's actually not true for the deployer. Can you remove the code and add an explicit comment? (the attributes are only used by the rails app)